src/serverSessionManager.ts

```diff
@@ -0,0 +1,124 @@
+    if ((debugSession.configuration as IPseudoAttachConfiguration).__pendingTargetId) {
+      return this.createChildDebugServer(debugSession);
+    } else {
+      return this.createRootDebugServer(debugSession, debugServerPort);
+    }
+  }
+
+  /**
+   * Create a new debug server for a new root session
+   * @param debugSession The IDE-specific debug session
+   * @returns The newly created debug server and a promise which resolves to the DapConnection associated with the session
+   */
+  public createRootDebugServer(
+    debugSession: T,
+    debugServerPort?: number,
+  ): IDebugServerCreateResult {
+    return this.innerCreateServer(
+      debugSession,
+      transport => this.sessionManager.createNewRootSession(debugSession, transport),
+      debugServerPort,
+    );
+  }
+
+  /**
+   * Create a new root debug session using an existing set of streams
+   * @param debugSession The IDE-specific debug session
+   * @param inputStream The DAP input stream
+   * @param outputStream The DAP output stream
+   */
+  public createRootDebugSessionFromStreams(
+    debugSession: T,
+    inputStream: Readable,
+    outputStream: Writable,
+  ): Session<T> {
+    const transport = new StreamDapTransport(inputStream, outputStream);
+    return this.sessionManager.createNewRootSession(debugSession, transport);
+  }
+
+  /**
+   * Create a new debug server for a new child session
+   * @param debugSession The IDE-specific debug session
+   * @returns The newly created debug server and a promise which resolves to the DapConnection associated with the session
+   */
+  public createChildDebugServer(debugSession: T): IDebugServerCreateResult {
+    return this.innerCreateServer(debugSession, transport =>
+      this.sessionManager.createNewChildSession(
+        debugSession,
+        (debugSession.configuration as IPseudoAttachConfiguration).__pendingTargetId,
+        transport,
+      ),
+    );
+  }
+
+  private innerCreateServer(
+    debugSession: T,
+    sessionCreationFunc: (transport: IDapTransport) => Session<T>,
+    debugServerPort?: number,
+  ): IDebugServerCreateResult {
+    const deferredConnection: IDeferred<DapConnection> = getDeferred();
+    const debugServer = net.createServer(socket => {
+      const transport = new StreamDapTransport(socket, socket);
+      const session = sessionCreationFunc(transport);
+      deferredConnection.resolve(session.connection);
+    });
+    debugServer.listen(debugServerPort || 0);
+    this.servers.set(debugSession.id, debugServer);
+
+    return { server: debugServer, connectionPromise: deferredConnection.promise };
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public terminate(debugSession: T) {
+    this.sessionManager.terminate(debugSession);
+    this.servers.get(debugSession.id)?.close();
+    this.servers.delete(debugSession.id);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public dispose() {
+    this.disposables.forEach(d => d.dispose());
```

Comment:
you only have one disposable in here

```suggestion
    this.sessionManager.dispose();
```